applications: asset_tracker: update README

bjda opened the pull request:
Brings asset tracker documentation in line with changes to master

Signed-off-by: Bernt Johan Damslora <bernt.johan.damslora@nordicsemi.no>

NordicBuilder commented:
**Some checks failed. Please fix and resubmit.**

## checkpatch (informational only, not a failure)

```
No additional types will be considered - file 'scripts/checkpatch/typedefsfile': No such file or directory
-:109: WARNING:TYPO_SPELLING: 'successfull' may be misspelled - perhaps 'successful'?
#109: FILE: applications/asset_tracker/README.rst:150:
+   #. If the pairing was successfull, the board and your nRF Cloud account are paired, and the device reboots.

- total: 0 errors, 1 warnings, 124 lines checked

NOTE: For some of the reported defects, checkpatch may be able to
      mechanically convert to the typical style using --fix or --fix-inplace.

Your patch has style problems, please review.

NOTE: Ignored message types: AVOID_EXTERNS BRACES CONFIG_EXPERIMENTAL CONST_STRUCT DATE_TIME FILE_PATH_CHANGES MINMAX MULTISTATEMENT_MACRO_USE_DO_WHILE NETWORKING_BLOCK_COMMENT_STYLE PRINTK_WITHOUT_KERN_LEVEL SPLIT_STRING VOLATILE

NOTE: If any of the errors are false positives, please report
      them to the maintainers.

```

## Gitlint issues

1: T5 Title contains the word 'wip' (case-insensitive): "WIP applications: asset_tracker: update README"
1: UC6 Body has no content, should at least have 1 line.



*Tip: The bot edits this comment instead of posting a new one, so you can check the comment's history to see earlier messages.*

umapraseeda commented on applications/asset_tracker/README.rst:
```diff
@@ -3,14 +3,15 @@
 ######################
 
-The Asset Tracker demonstrates how to use the :ref:`lib_nrf_cloud` to connect an nRF9160 DK to the `nRF Cloud`_ via LTE, transmit GPS and sensor data, and retrieve information about the modem.
+The Asset Tracker demonstrates how to use the :ref:`lib_nrf_cloud` to connect a nRF9160-based board to the `nRF Cloud`_ via LTE, transmit GPS and sensor data, and retrieve information about the device.
```
```
	Suggestion: connect an nRF9160-based board
````

bjda replied:
Implemented

umapraseeda commented on applications/asset_tracker/README.rst:
```diff
@@ -29,21 +30,21 @@ This data includes the GPS position, accelerometer data (the device's physical o
      - AIR_PRESS
 
-By default, the application uses simulated sensor data, but it can be configured with Kconfig options to use real sensors to collect data.
+On the nRF9160 DK, the application uses simulated sensor data by default, but it can be configured with Kconfig options to use real sensors to collect data. On the Thingy:91, onboard sensors are used by default. GPS is enabled by default on both boards.
```
```
	Suggestion: GPS is enabled by default on both the boards.
```

bjda replied:
Implemented

umapraseeda commented on applications/asset_tracker/README.rst:
```diff
@@ -52,16 +53,19 @@ Requirements
 
-The two buttons and two switches are used to enter a pairing pattern to associate a specific board with an nRF Cloud user account.
-In addition, the two switches have the following function when the connection is established:
+The buttons and switches have the following function when the connection is established:
```
```
Suggestion:  The buttons and switches have the following functions when the connection is established:
```

bjda replied:
Implemented

umapraseeda commented on applications/asset_tracker/README.rst:
```diff
@@ -52,16 +53,19 @@ Requirements
+
+Button 1:
+    * Send a BUTTON event to nRF Cloud.
+    * (10 s long press) Toggle GPS function
```
```	
 Suggestion: Toggle GPS function ( long press the button for a minimum of 10 seconds)
```

bjda replied:
Implemented

umapraseeda commented on applications/asset_tracker/README.rst:
```diff
@@ -132,22 +141,22 @@ After programming the application and all prerequisites to your board, test the
       ***** Booting Zephyr OS v1.13.99 *****
```
```
Suggestion: Don't we have to reset the board to see this message? If that is the case may be we should add a step here to reset the board as step 4. 
```

bjda replied:
Implemented, added as additional step

umapraseeda commented on applications/asset_tracker/README.rst:
```diff
@@ -132,22 +141,22 @@ After programming the application and all prerequisites to your board, test the
+#. Press button 1 to send BUTTON data to nRF Cloud.
+#. Press button 1 for at least 10 seconds to enable GPS tracking.
+   This requires a good view of the sky and up to a few minutes to get the first position fix.
 #. Optionally send AT commands from the terminal, and observe that the reponse is received.
```
````
 Suggestion: and observe that the response
````

bjda replied:
Implemented

thst-nordic commented:
you can rebase master to get rid of the 'continuous-integration/jenkins/pr-head' failure.

bjda commented:
Changes since last review:
* Implemented all suggestions
* Inserted missing sections
* Removed reference to button+switch pairing
* Added light sensor to table
* Added firmware update library reference
* Moved web step of instruction
* Removed redundant LED step of instruction
* Consistently use "nRF Cloud", rather than sometimes "the nRF Cloud"

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -52,16 +53,19 @@ Requirements
 
-Switch 1:
+Switch 1 (only on nRF9160 DK):
     * Toggle to simulate flipping of the boards orientation.
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -132,22 +141,22 @@ After programming the application and all prerequisites to your board, test the
-#. Toggle switch 1 to simulate flipping the board orientation.
-#. Set switch 2 in the position marked **N.C.** and observe that simulated GPS data is sent to the nRF Cloud.
+#. Press button 1 to send BUTTON data to nRF Cloud.
+#. Press button 1 for at least 10 seconds to enable GPS tracking.
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -132,22 +141,22 @@ After programming the application and all prerequisites to your board, test the
-#. Set switch 2 in the position marked **N.C.** and observe that simulated GPS data is sent to the nRF Cloud.
+#. Press button 1 to send BUTTON data to nRF Cloud.
+#. Press button 1 for at least 10 seconds to enable GPS tracking.
+   This requires a good view of the sky and up to a few minutes to get the first position fix.
```
Implemented

jtguggedal commented on applications/asset_tracker/README.rst:
```diff
@@ -52,19 +57,22 @@ Requirements
+
+Button 1:
+    * Send a BUTTON event to nRF Cloud.
+    * Toggle GPS function ( long press the button for a minimum of 10 seconds)
```
Suggestion: `Toggle GPS operation`
Also, an extra space seems to be present after `(`

bjda replied:
Implemented

jtguggedal commented on applications/asset_tracker/README.rst:
```diff
@@ -77,28 +85,41 @@ LED 3 and LED 4:
+   "Yellow", "Waiting for user association"
+   "Blue", "Connected, sending environment data"
+   "Purple", "Searching for GPS"
+   "Green", "Connected, sending GPS and environment data"
```
I think green indicates GPS fix as well

bjda replied:
Added "GPS has fix"

jtguggedal commented on applications/asset_tracker/README.rst:
```diff
@@ -77,28 +85,41 @@ LED 3 and LED 4:
 ******************
 
-The Asset Tracker can run in three power modes that are configured in the Kconfig file of the application.
+The Asset Tracker can run in three power modes that are configured in the Kconfig file of the application. These settings are currently only supported on the nRF9160 DK.
```
To be a bit picky, Thingy:91 also runs  in PSM when the GPS has started

bjda replied:
It does. However, I don't think that detail belongs in the power optimization section. The device isn't really saving power in that case, as it is using the radio for GPS.

jtguggedal commented on applications/asset_tracker/README.rst:
```diff
@@ -77,28 +85,41 @@ LED 3 and LED 4:
 
 Request eDRX mode
-	In this mode, the device sends GPS data every 2 minutes.
+	In this mode, the device requests the eDRX feature from the cellular network to save power.
```
Perhaps add something like `Note that eDRX is not enabled by all network providers, and the granted eDRX parameters may be different from what is requested.`

bjda replied:
Added to section intro

jtguggedal commented on applications/asset_tracker/README.rst:
```diff
@@ -77,28 +85,41 @@ LED 3 and LED 4:
+	In this mode, the device requests the eDRX feature from the cellular network to save power.
 	To enable this mode, set ``CONFIG_POWER_OPTIMIZATION_ENABLE=y`` and then
-	set Switch 2 to ON.
+	set Switch 2 to N.C.
```
I'd add `N.C. position` to be super-explicit

bjda replied:
Added

jtguggedal commented on applications/asset_tracker/README.rst:
```diff
@@ -77,28 +85,41 @@ LED 3 and LED 4:
-	set Switch 2 to OFF.
+	In this mode, the device requests the PSM feature from the cellular network to save power.
+	To enable this mode, set ``CONFIG_POWER_OPTIMIZATION_ENABLE=y`` and then
+	set Switch 2 to GND.
```
Same note as for eDRX may be relevant here. Depends on how detailed we should be on explaining these features in the application docs, though.

bjda replied:
Added to section intro

jtguggedal commented on applications/asset_tracker/README.rst:
```diff
@@ -121,33 +142,32 @@ After programming the application and all prerequisites to your board, test the
 
-#. Observe that LED 3 starts blinking as the LTE link is established. This may take several minutes.
-#. Observe in the terminal window that the connection to nRF Cloud is established.
+#. Observe in the terminal window that the connection to nRF Cloud is established. This may take several minutes.
```
Should we keep a mention of the LTE link here? `Observe in the terminal window that LTE link and connection to nRF cloud is established`

bjda replied:
Since the LTE link notification is followed by more scrolling text quite soon, I don't think it makes sense to ask the user to look for it.

umapraseeda commented on applications/asset_tracker/README.rst:
```diff
@@ -121,33 +153,32 @@ After programming the application and all prerequisites to your board, test the
-#. Set switch 2 in the position marked **N.C.** and observe that simulated GPS data is sent to the nRF Cloud.
-#. Optionally send AT commands from the terminal, and observe that the reponse is received.
+#. Press button 1 to send BUTTON data to nRF Cloud.
+#. Press button 1 for a minimum of 10 seconds to enable GPS tracking.
```
Suggestion: Should we mention SW3 in the case of Thingy specifically?

bjda replied:
Added in parentheses.

umapraseeda commented on applications/asset_tracker/README.rst:
```diff
@@ -77,28 +85,52 @@ LED 3 and LED 4:
 ******************
 
-The Asset Tracker can run in three power modes that are configured in the Kconfig file of the application.
+The Asset Tracker can run in three power modes that are configured in the Kconfig file of the application. These settings are currently only supported on the nRF9160 DK. Note that not all cellular network providers support these modes, and the granted parameters can vary between networks.
```
```
Suggestion: Should we add the note as follows?
.. note::
   Not all cellular network providers support these modes, and the granted parameters can vary between networks.
 ```

bjda replied:
Implemented

ru-fu commented on applications/asset_tracker/README.rst:
```diff
@@ -121,33 +153,32 @@ After programming the application and all prerequisites to your board, test the
-#. Optionally send AT commands from the terminal, and observe that the reponse is received.
+#. Press button 1 (SW3 on Thingy:91) to send BUTTON data to nRF Cloud.
+#. Press button 1 (SW3 on Thingy:91) for a minimum of 10 seconds to enable GPS tracking.
+   The board has to be outdoors in clear space for a few minutes to get the first position fix.
```
"must be" instead of "has to be"

bjda replied:
Fixed

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -28,22 +29,26 @@ This data includes the GPS position, accelerometer data (the device's physical o
 
-In addition to the sensor data, the application retrieves information about the LTE modem, such as the signal strength, battery voltage, and current operator.
-This information is available in nRF Cloud under the sensor type "DEVICE".
+The LTE Link Monitor PC tool can be used to send AT commands and receive the response.
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -52,19 +57,22 @@ Requirements
+    * Toggle to simulate orientation change (flipping) of the board.
 
-The application state is indicated by the LEDs.
+Switch 2 (only on NRF9160 DK):
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -52,19 +57,22 @@ Requirements
 
-Switch 1:
-    * Toggle to simulate flipping of the boards orientation.
+Button 1:
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -121,33 +153,32 @@ After programming the application and all prerequisites to your board, test the
-      If the LEDs start blinking in pairs, check in the terminal window which error occurred.
+   a. Observe that the LED(s) indicate that the device is waiting for user association.
+   #. Follow the instructions on `nRF Cloud`_ to pair your device.
+   #. If the pairing was successful, the board and your nRF Cloud account are paired, and the device reboots.
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -121,33 +153,32 @@ After programming the application and all prerequisites to your board, test the
+   a. Observe that the LED(s) indicate that the device is waiting for user association.
+   #. Follow the instructions on `nRF Cloud`_ to pair your device.
+   #. If the pairing was successful, the board and your nRF Cloud account are paired, and the device reboots.
+      If the LED(s) indicate an error, check in the terminal window which error occurred.
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -121,33 +153,32 @@ After programming the application and all prerequisites to your board, test the
-#. Toggle switch 1 to simulate flipping the board orientation.
-#. Set switch 2 in the position marked **N.C.** and observe that simulated GPS data is sent to the nRF Cloud.
-#. Optionally send AT commands from the terminal, and observe that the reponse is received.
+#. Press button 1 (SW3 on Thingy:91) to send BUTTON data to nRF Cloud.
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -121,33 +153,32 @@ After programming the application and all prerequisites to your board, test the
-#. Set switch 2 in the position marked **N.C.** and observe that simulated GPS data is sent to the nRF Cloud.
-#. Optionally send AT commands from the terminal, and observe that the reponse is received.
+#. Press button 1 (SW3 on Thingy:91) to send BUTTON data to nRF Cloud.
+#. Press button 1 (SW3 on Thingy:91) for a minimum of 10 seconds to enable GPS tracking.
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -3,14 +3,15 @@
 ######################
 
-The Asset Tracker demonstrates how to use the :ref:`lib_nrf_cloud` to connect an nRF9160 DK to the `nRF Cloud`_ via LTE, transmit GPS and sensor data, and retrieve information about the modem.
+The Asset Tracker demonstrates how to use the :ref:`lib_nrf_cloud` to connect an nRF9160-based board to `nRF Cloud`_ via LTE, transmit GPS and sensor data, and retrieve information about the device.
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -52,19 +57,22 @@ Requirements
-Switch 1:
-    * Toggle to simulate flipping of the boards orientation.
+Button 1:
+    * Send a BUTTON event to nRF Cloud.
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -52,19 +57,22 @@ Requirements
 
 LED 3 and LED 4:
-    * LED 3 blinking: Connecting - The device is resolving DNS and connecting to the nRF Cloud.
+    * LED 3 blinking: Connecting - The device is resolving DNS and connecting to nRF Cloud.
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -77,28 +85,52 @@ LED 3 and LED 4:
 All LEDs (1-4):
     * Blinking in groups of two (LED 1 and 3, LED 2 and 4): Recoverable error in the BSD library.
-    * Blinking in cross pattern (LED 1 and 4, LED 2 and 3): Communication error with the nRF Cloud.
+    * Blinking in cross pattern (LED 1 and 4, LED 2 and 3): Communication error with nRF Cloud.
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -77,28 +85,52 @@ LED 3 and LED 4:
+   * - White
+     - Connecting to network
+   * - Cyan
+     - Connecting to nRF Cloud
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -121,33 +153,32 @@ After programming the application and all prerequisites to your board, test the
 
-#. Observe that LED 3 starts blinking as the LTE link is established. This may take several minutes.
-#. Observe in the terminal window that the connection to nRF Cloud is established.
+#. Observe in the terminal window that the connection to nRF Cloud is established. This may take several minutes.
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -121,33 +153,32 @@ After programming the application and all prerequisites to your board, test the
       The device must be power-cycled to restart the pairing procedure.
-   #. After reboot, the board connects to the nRF Cloud, and the pattern disappears from the web page.
-#. Observe that LED 4 is turned on to indicate that the connection is established.
+   #. After reboot, the board connects to nRF Cloud.
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -121,33 +153,32 @@ After programming the application and all prerequisites to your board, test the
-#. Toggle switch 1 to simulate flipping the board orientation.
-#. Set switch 2 in the position marked **N.C.** and observe that simulated GPS data is sent to the nRF Cloud.
-#. Optionally send AT commands from the terminal, and observe that the reponse is received.
+#. Press button 1 (SW3 on Thingy:91) to send BUTTON data to nRF Cloud.
```
Implemented

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -3,14 +3,15 @@
 
-The application sends data that is collected by an nRF9160 DK to Nordic Semiconductor's cloud solution, `nRF Cloud`_, where the data is visualized.
-This data includes the GPS position, accelerometer data (the device's physical orientation), and data from various environment sensors.
+The application uses the LTE link control driver to establish a network connection. It then collects various data locally, and transmits the data to Nordic Semiconductor's cloud solution, `nRF Cloud`_. The data is visualized in nRF Cloud's web interface.
```
Implemented. I assume this applies to all prose paragraphs.

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -28,22 +29,26 @@ This data includes the GPS position, accelerometer data (the device's physical o
 
-The LTE link control driver is used to establish the LTE link automatically.
-When the device is connected to the nRF Cloud, you can use the LTE Link Monitor to send AT commands and receive the response.
+By default, the asset tracker supports firmware updates through :ref:`lib_aws_fota`.
```
I agree this is too thin. Since all the logic relating to updates is implemented in and dictated by the FOTA library (it is only enabled in the asset tracker), I think that is a better place for most of the technical details. I agree that how to update the firmware through nRF Cloud from a user perspective should be documented somewhere, and probably referenced from here.

This will have to be done after release, as the behavior is not yet quite stable in nRF Cloud yet, AFAIK.

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -52,19 +57,22 @@ Requirements
-    * Toggle to simulate flipping of the boards orientation.
+Button 1:
+    * Send a BUTTON event to nRF Cloud.
+    * Toggle GPS operation (long press the button for a minimum of 10 seconds).
```
Yes, that is valid for both boards. The intention is that all text is valid for all boards, unless otherwise noted.

The dictionary definition of toggle (v.) was more vague than I thought. Edited to be more precise.

Simulated GPS operation is not enabled by default on any board, and is not mentioned in the documentation.

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -77,28 +85,52 @@ LED 3 and LED 4:
+   :header-rows: 1
+   :align: center
+
+   * - LED Color
```
Fixed

bjda commented on applications/asset_tracker/README.rst:
```diff
@@ -121,33 +153,32 @@ After programming the application and all prerequisites to your board, test the
 1. Connect the board to the computer using a USB cable.
```
Yes, that is valid for both boards. The intention is that all text is valid for all boards, unless otherwise noted.

Connecting the Thingy:91 to a terminal emulator entails the same operations as with the DK.

bjda commented:
Changes since last review:
* Implemented all suggestions, incl. consistently use "the nRF Cloud", rather than "nRF Cloud"
* Clarified "toggle" to "enable or disable"

ru-fu commented:
This PR needs a backport PR - @thst-nordic or @rlubos , who should do this?

rlubos commented:
@ru-fu The bot should do this, the question is who **will** do this :)

rlubos commented:
I can open a backport PR, just give me a sec.
